types/lodash/ts3.1/common/lang.d.ts

```diff
@@ -838,7 +838,7 @@ declare module "../index" {
         isMap(): PrimitiveChain<boolean>;
     }
 
-    type isMatchCustomizer = (value: any, other: any, indexOrKey?: PropertyName) => boolean;
+    type isMatchCustomizer = (value: any, other: any, indexOrKey?: PropertyName) => boolean | undefined;
```

Comment:
It looks like this type is never used. Can we delete it instead?

Same for the non-3.1 version.

Reply:
It is used, for `isMatchWith`

Reply:
`isMatchWith(object: object, source: object, customizer: isMatchWithCustomizer): boolean`

Reply:
That is `isMatchWithCustomizer`, not `isMatchCutsomizer`.

Reply:
Oh, true. That's what I updated in my project when testing. Will update the `isMatchWithCustomizer` and remove this one

Reply:
But seems like it's used in v3
    //_.isMatch
    interface isMatchCustomizer {
        (value: any, other: any, indexOrKey?: number|string): boolean;
    }

    interface LoDashStatic {
        /**
         * Performs a deep comparison between object and source to determine if object contains equivalent property
         * values. If customizer is provided it’s invoked to compare values. If customizer returns undefined
         * comparisons are handled by the method instead. The customizer is bound to thisArg and invoked with three
         * arguments: (value, other, index|key).
         * @param object The object to inspect.
         * @param source The object of property values to match.
         * @param customizer The function to customize value comparisons.
         * @param thisArg The this binding of customizer.
         * @return Returns true if object is a match, else false.
         */
        isMatch(object: Object, source: Object, customizer?: isMatchCustomizer, thisArg?: any): boolean;
    }

    interface LoDashImplicitObjectWrapper<T> {
        /**
         * @see _.isMatch
         */
        isMatch(source: Object, customizer?: isMatchCustomizer, thisArg?: any): boolean;
    }

Reply:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/lodash/v3/index.d.ts#L11287-L11312

Reply:
Oh I see, that explains where it came from.

In that case, let's:
-Remove `isMatchCustomizer` from this file
- Also remove it from `lodash/common/lang.d.ts`
- Fix it in v3 (and add/update a test if possible)

Reply:
Why is there no `isMathcWith` in `/v3/index.d.ts`? https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/lodash/v3/index.d.ts#L11287-L11312

Reply:
@aj-r

Reply:
Oh, it's correct. There was no `isMatchWith` in v3.1. It was separated in v4

Reply:
Is /ts3.1 - version of types for `lodash` 4.0+?